### PR TITLE
Allow iptool to function even when we can't determine local interfaces

### PR DESCRIPTION
**Pull request by oxtoacart**
For getlantern/android-lantern#197

### CODE REVIEW

File: iptool.go [hunk 1]
```diff
@@ -57,33 +56,42 @@ func init() {
...
-		return nil, errors.New("Unable to determine interface addresses: %v", err)
-	}
-	privateNets := make([]*net.IPNet, len(globalPrivateUseNets), len(globalPrivateUseNets)+len(addrs))
-	copy(privateNets, globalPrivateUseNets)
-	for _, addr := range addrs {
-		switch t := addr.(type) {
-		case *net.IPNet:
-			privateNets = append(privateNets, t)
+		// This can happen on Android due to https://github.com/golang/go/issues/40569
+		log.Errorf("Unable to determine interface addresses, will proceed with using only global private-use networks: %v", err)
```
Comment (oxtoacart) [inline iptool.go:81]: This is the main fix. Instead of returning an error, just log the error and continue to initialize the Tool without using local interface addresses.

File: iptool.go [hunk 2]
```diff
@@ -4,12 +4,11 @@ package iptool
 import (
 	"net"
 
-	"github.com/getlantern/errors"
 	"github.com/getlantern/golog"
 )
 
 var (
-	log = golog.LoggerFor("landetect")
+	log = golog.LoggerFor("iptool")
```
Comment (oxtoacart) [inline iptool.go:11]: Drive-by: logger name should match package

File: iptool.go [hunk 3]
```diff
@@ -57,33 +56,42 @@ func init() {
 
 type Tool interface {
 	// IsPrivate checks whether the given IP address is private, meaning it's
-	// using one of addresses designed by IANA as not routable on the Internet or
-	// the address of one of the interfaces on the current host.
+	// using one of the addresses designated by IANA as not routable on the Internet,
```
Comment (oxtoacart) [inline iptool.go:59]: Drive-by: grammar/spelling

File: iptool_test.go [hunk 4]
```diff
@@ -5,21 +5,18 @@ import (
...
+	"github.com/stretchr/testify/require"
 )
 
 func TestIsPrivate(t *testing.T) {
-	tool, err := New()
-	if !assert.NoError(t, err) {
-		return
-	}
+	tool, includesLocalInterfaces := New()
+	require.True(t, includesLocalInterfaces)
```
Comment (oxtoacart) [inline iptool_test.go:13]: Drive-by: require is clearer than assert

Comment (Crdzbird) [reply iptool_test.go:13]: Im not sure with replacing `assert` with `require` if we keep require then if an error on a step of the test happens then the while test is gonna be skipped, contrary to assert which logs each test and if fails then stops on the one that failed. But please feel free to correct me if im wrong or it's behavior is different.

Comment (oxtoacart) [reply iptool_test.go:13]: Yeah, this would actually be a good place for an assert. I'll change, thanks!

I tend to use require when checking for errors, since there's little point in continuing if we encountered an error.

Comment (oxtoacart) [reply iptool_test.go:13]: Oh actually, now that I think about it, if includesLocalInterfaces is false, there are later test conditions that will also fail because the test assumes that local interfaces are being included. So, I'll stick with `require` here :)

**Comment by oxtoacart** [comment]
Thanks for the review @Crdzbird !
